core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpLeonardoDAO.scala

```diff
@@ -17,11 +17,19 @@ class HttpLeonardoDAO(leonardoConfig: LeonardoConfig) extends LeonardoDAO {
     new AppsV2Api(apiClient)
   }
 
-  override def createWDSInstance(token: String, workspaceId: UUID): Unit =
-    createApp(token, workspaceId, s"wds-$workspaceId", leonardoConfig.wdsType)
+  override def createWDSInstance(token: String, workspaceId: UUID, sourceWorkspaceId: Option[UUID] = None): Unit =
+    createApp(token, workspaceId, s"wds-$workspaceId", leonardoConfig.wdsType, sourceWorkspaceId)
 
-  override def createApp(token: String, workspaceId: UUID, appName: String, appType: String): Unit = {
+  override def createApp(token: String,
+                         workspaceId: UUID,
+                         appName: String,
+                         appType: String,
+                         sourceWorkspaceId: Option[UUID]
+  ): Unit = {
     val createAppRequest = new CreateAppRequest()
+    if (sourceWorkspaceId.nonEmpty) {
+      createAppRequest.setSourceWorkspaceId(sourceWorkspaceId.get.toString)
+    }
     val appTypeEnum = AppType.fromValue(appType)
     createAppRequest.setAppType(appTypeEnum)
     getAppsV2leonardoApi(token).createAppV2(workspaceId.toString, appName, createAppRequest)
```

Comment:
for unit testing, I recommend breaking out the logic to construct the `CreateAppRequest` into a standalone unit-testable method, something like this:
```scala
  override def createApp(token: String,
                         workspaceId: UUID,
                         appName: String,
                         appType: String,
                         sourceWorkspaceId: Option[UUID]
                        ): Unit = {
    val createAppRequest = buildAppRequest(appType, sourceWorkspaceId)
    getAppsV2leonardoApi(token).createAppV2(workspaceId.toString, appName, createAppRequest)
  }

  protected[dataaccess] def buildAppRequest(appType: String, sourceWorkspaceId: Option[UUID]): CreateAppRequest = {
    val createAppRequest = new CreateAppRequest()
    sourceWorkspaceId.foreach { sourceId =>
      createAppRequest.setSourceWorkspaceId(sourceId.toString)
    }
    val appTypeEnum = AppType.fromValue(appType)
    createAppRequest.setAppType(appTypeEnum)

    createAppRequest
  }
```

Then, you can unit test directly against `buildAppRequest` with different values for `sourceWorkspaceId`.

Note: I'm not sure if you need the `protected[dataaccess]` on `buildAppRequest` - just guessing without actually trying it what visibility is best for being unit-testable from `HttpLeonardoDAOSpec`